src/pages/billing/Project.js

```diff
@@ -145,23 +152,38 @@ const ProjectDetail = ({ project, project: { projectName, creationStatus }, bill
 
   const adminCanEdit = _.filter(({ roles }) => _.includes(billingRoles.owner, roles), projectUsers).length > 1
 
+  const workspacesInProject = useMemo(() => _.flow(
+    _.map(({ workspace }) => workspace),
+    _.filter({ namespace: projectName })
+  )(workspaces), [workspaces, projectName])
+
+  const getBillingAccountStatusIcon = (() => {
+    const allWorkspacesHaveCorrectBillingAccount =
+      _.isEmpty(_.remove({ billingAccount: billingAccountName }, workspacesInProject))
+    const getIcon = shape => icon(shape, { size: workspaceStatusIconWidth })
+    return workspace => allWorkspacesHaveCorrectBillingAccount ? paddingForWorkspaceStatusIcon :
+      billingAccountName === workspace.billingAccount ? getIcon('success-standard') :
+        'billingAccountErrorMessage' in workspace ? getIcon('error-standard') :
+          getIcon('loadingSpinner')
+  })()
```

Comment:
We have a utility function to handle these types of conditionals:

```suggestion
  const statusIcon = Utils.cond(
    [_.includes({ billingAccount: billingAccountName }, workspace), () => null],
    [billingAccountName === workspace.billingAccount, () => 'success-standard'],
    ['billingAccountErrorMessage' in workspace, () => 'error-standard'],
    [Utils.DEFAULT, () => 'loadingSpinner']
  )
```

Move this above line 79 where I annotated above.

Before this change I was seeing busy spinners on my accounts. Please double check the functionality to make sure the correct icons are appearing.

Reply:
Thanks for `cond` - I was hoping we had something like that instead of ternaries!
The spinners are because of an issue in the UI this change exposes (also, it only works when using rawls v2_workspaces). I'm going to fix it in a subsequent change that needs to be double submitted to here and dev. The ticket is https://broadworkbench.atlassian.net/browse/CA-1491